qlib/WSDL.qm

```diff
@@ -3057,41 +3137,46 @@ public class WSDL::SoapBinding inherits WSDL::OperationalBinding {
         # remove namespace prefixes from body element keys
         WSDL::XsdBase::removeNS2(\mrh.body);
 
-        if (body.multiRef) {
-            foreach auto mr in (body.multiRef)
-                mrh.mrh.(mr."^attributes^".id) = mr;
-
-            # resolve interior references to multiRefs
-            foreach string id in (keys mrh.mrh) {
-                foreach string key in (keys mrh.mrh{id}) {
-                    if (key == "^attributes^")
-                        continue;
-
-                    #printf("multiRef id: %y key: %y val: %y\n", id, key, mrh.mrh{id}{key});
-
-                    if (mrh.mrh{id}{key}.typeCode() == NT_LIST) {
-                        foreach auto e in (\mrh.mrh{id}{key}) {
-                            auto href = substr(e."^attributes^".href, 1);
-                            if (exists href) {
-                                if (!exists mrh.mrh{href})
-                                    throw "INVALID-MULTIREF", sprintf("multiRef id: %y does not exist", href);
-                                e = mrh.mrh{href};
-                            }
-                        }
-                    }
-                    else if (mrh.mrh{id}{key}.typeCode() == NT_HASH) {
-                        auto href = substr(mrh.mrh{id}{key}."^attributes^".href, 1);
+        # issue #2902: get all multiref elements with an "id" tag
+        foreach hash<auto> hh in (body.pairIterator()) {
+            if (!exists hh.value."^attributes^".id) {
+                continue;
+            }
+            hash<auto> h = remove body{hh.key};
+            string id = remove h."^attributes^".id;
+            # FIXME: process root & type
```

Comment:
is there an issue opened to address this FIXME?

Reply:
yes, qorelanguage/qore#2902